Pre-requisites script

XhorxhinaT opened the pull request:
This script checks if the needed libraries and packages are installed, so our tool can run without errors.

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
We need to ease the installation of the tool to the user. We are already installing libraries and dependencies and we need to take care of them to facilitate the installation of the user.
We need a script that:

Checks that Google SDK is installed, if not print a link to show where to download it / how to install it. If it is, configure the project written by the user in the config file.
Check if geth and puppeth(docker also needed to be installed) are installed, if not print a link to show how to install it.
Check if jq (used in the SUT script) is installed, if not print a link to show how to install it.
Check if npm is installed. If not print a link how to install it.
Check if caliper is installed under /bin. If not, it will be installed with npm.
Run caliper bind to configure ethereum.
Execute the template creation script #61
If possible in this script, check that the python libraries needed are installed. If possible find an easy way to install these libraries. If it is not possible it should be enough with checking if these are installed or not.
## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->




## Related to
#30 



## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Checks the libraries and packages, if everything is installed or not. 


## Checklist contributor:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] The code of the script runs without any errors (error handling correctly used)

- [x] The code should print clear steps of how to install or at least show a link with clear documentation of how to install what is needed.

- [x] If everything is installed, show a message that dependencies are fulfilled. 
- [ ] If after running the script everything is fulfilled, main.py must be able to run correctly without dependencies errors

<!--- source: https://www.talater.com/open-source-templates/# -->

XhorxhinaT committed:
Pre-requisites script

This script checks if the needed libraries and packages are installed, so our tool can run without errors.

XhorxhinaT committed:
Update prerequisites.sh

XhorxhinaT commented:
jj

XhorxhinaT committed:
Update prerequisites.sh

fayanpardo commented on prerequisites.sh:
```diff
@@ -0,0 +1,102 @@
+
+cd cp_ws_1920/bin
+
+if hash "caliper" 2>/dev/null; then
```
I think you should check first if npm is installed before checking caliper because you need npm to install caliper.

XhorxhinaT replied:
right! thx

fayanpardo commented on prerequisites.sh:
```diff
@@ -0,0 +1,102 @@
+
+ if hash "gcloud" 2>/dev/null; then
+    echo 'Google cloud SDK is installed. Please wait a few seconds..'
+    gcloud auth configure-docker
```
instead of doing this command, can we add the command to configure a gcp project? In the task is asked to configure the project the user will write in the config file.

SantoshDhirwani replied:
I think you will need both:

gcloud auth login

gcloud config set project my-project

XhorxhinaT replied:
maybe i could live it up to the User to do that command "gcloud config set project my-project"
The project ID changes for everyone, unless the other task where the Proect ID is asked, can be submitted in this script here.

fayanpardo commented on prerequisites.sh:
```diff
@@ -0,0 +1,102 @@
+  echo "Let's run the template 61-skript-TODO"
+ # ../Template61- I need to know the path
+ echo ""
+  echo "Pre-requisites are all checked. Please see if something is missing and needs to be installed."
```
You should show a success message if everything is installed or working correctly. Maybe what you can do is to have a boolean variable from the beginning and if something is not installed then change this variable to false. In the end you can check this variable to print a failure or success value to the user ;)

XhorxhinaT replied:
I tried that already, and bash is absurd sometimes. Like i tried this for example, and changed the and and or, and the logic doesnt seem to go well. maybe i got dull
if aa=$1 && bb= $1 ; then
if hash "geqth" 2>/dev/null; then
	
	 aa= $1
    echo 'geth is installed'
  else
    echo 'geth is missing. Please follow this link https://github.com/ethereum/go-ethereum/wiki/Installing-Geth to install geth'
   
  
 
  
 fi 
  if hash "puppeeth" 2>/dev/null; then
	    bb= $1
    echo 'puppeth is installed'
  else 
    echo 'puppeth is missing. Please follow this link https://www.sitepoint.com/puppeth-introduction/ to install puppeth'
  
    echo " $aa is not good"
  fi
echo "  ok";

else
	echo "noo okkkk"
fi

fayanpardo reviewed (changes requested):
Nice job! Happy you found how to check for python libraries. There are some minor changes to do from the task still :)

XhorxhinaT committed:
Update prerequisites.sh

XhorxhinaT committed:
Updated Prerequisities

I still got to add the path of Template#61 -TODO after I have the script.
Thinking a way to control the final echo when they all are installed (&&)=true
or with (-o) =true

XhorxhinaT committed:
Finished Pre-requisities

fayanpardo commented on prerequisites.sh:
```diff
@@ -0,0 +1,121 @@
+  fi
+  
+  echo ""
+   cd cp_ws_1920/bin
```
error here! wrong path

XhorxhinaT replied:
why wrong path?

XhorxhinaT replied:
and what comment did i left out? :/

XhorxhinaT replied:
@boxrhcp wrong path? why is bin not undeer cp_ws

fayanpardo replied:
I get the following errors.
![image](https://user-images.githubusercontent.com/12150712/72884907-36b33100-3d07-11ea-89e2-b2de01d66dc9.png)

bin is under cp_ws but im running the script inside cp_ws folder.

XhorxhinaT replied:
<img width="960" alt="Screen Shot 2020-01-22 at 19 50 47" src="https://user-images.githubusercontent.com/56897059/72924128-adc0e780-3d50-11ea-8af6-abeff49c96af.png">
 corrected that one thing

XhorxhinaT replied:
@boxrhcp gracias

fayanpardo reviewed (changes requested):
Now it prompts a correct message but you left behind what is commented

XhorxhinaT committed:
Prerequisities done

fayanpardo reviewed (dismissed):
GJ!

SantoshDhirwani commented on prerequisites.sh:
```diff
@@ -0,0 +1,121 @@
+    echo 'numpy is installed'
+fi
+  echo ""
+  echo "Let's run the template 61-skript-TODO"
```
Where are you calling the script from #61 ? 
It seems like it is still a TODO.

santoshD93 committed:
Merge branch 'master' into XhorxhinaT-patch-1

santoshD93 committed:
added calling script for creating VM template, organized structure and tested

SantoshDhirwani reviewed (dismissed):
It works now. GJ!

SantoshDhirwani commented:
> @SantoshDhirwani did u merge it after did the change?

No, your branch was behind the master so I just brought it on the same level as master without losing your changes, or else your script would not be able to find my scripts from #61.
You can test this branch now and it should work.

XhorxhinaT committed:
finished Prerequsities

The last ToDo also runs well

XhorxhinaT reviewed (commented):
Now it's finished! Approve please @boxrhcp @ulalaa14 @santosh-dhirwani @ankushshr1993

boxrhcp commented:
Did you test it? the script is under /bin not in the base directory and is not checking if it worked or not :/

XhorxhinaT commented:
> Did you test it? the script is under /bin not in the base directory and is not checking if it worked or not :/

yes i did, santosh part worked, but caaliper didn't, and that comes because the script in my branch ended somehow under bin, which means, I have to refix caliper, or move the script.  Your want prereq our of bin right?

XhorxhinaT committed:
fixed_prereq

prereq.sh is out of bin now and caliper and template should work

SantoshDhirwani commented:
As a side note: pip install pandas doesn't work for mac (at least it didn't work for me). So my question is that will these pre-requisites work for all types of operating systems? If not, we must mention it clearly in the readme.

XhorxhinaT commented:
@santosh-dhirwani  it looks like it works for me, and I have mac. Maybe you should upgrade your pip or use instead pip3
(base) MAC-UM-265:~ xhorxhina.taraj$   pip install pandas
Requirement already satisfied: pandas in ./opt/anaconda3/lib/python3.7/site-packages (0.25.1)
Requirement already satisfied: python-dateutil>=2.6.1 in ./opt/anaconda3/lib/python3.7/site-packages (from pandas) (2.8.0)
Requirement already satisfied: pytz>=2017.2 in ./opt/anaconda3/lib/python3.7/site-packages (from pandas) (2019.3)
Requirement already satisfied: numpy>=1.13.3 in ./opt/anaconda3/lib/python3.7/site-packages (from pandas) (1.17.2)
Requirement already satisfied: six>=1.5 in ./opt/anaconda3/lib/python3.7/site-packages (from python-dateutil>=2.6.1->pandas) (1.12.0)

SantoshDhirwani commented:
Okay, great. I didn't check that on mac because I also switched to working in a VM (Ubuntu) recently.

SantoshDhirwani reviewed (approved):
GJ!
